src/charm.py

```diff
@@ -189,6 +202,9 @@ def _common_exit_hook(self) -> None:
         # Update config file
         try:
             self.blackbox_workload.update_config()
+            modules_from_relation = self._probes_consumer.modules()
+            if modules_from_relation:
+                self._update_blackbox_config_yaml_from_relation(modules_from_relation)
```

Comment:
Should we make this cleaner:
```python
self._update_blackbox_config_yaml_from_relation(self._probes_consumer.modules())
```
I.e. `.modules()` could always return an iterable.